synapse/server_notices/server_notices_manager.py

```diff
@@ -288,3 +325,51 @@ async def _update_notice_user_profile_if_changed(
                 ratelimit=False,
                 content={"displayname": display_name, "avatar_url": avatar_url},
             )
+
+    async def _update_room_info(
+        self,
+        requester: Requester,
+        room_id: str,
+        info_event_type: str,
+        info_content_key: str,
+        info_value: Optional[str],
+        if_changed: Optional[bool] = True,
```

Comment:
I find this name a bit confusing, I'd probably call it `check_if_changed`. It also doesn't need to be `Optional`. 

But since we don't use it anywhere I'd suggest just removing it entirely.

Reply:
Oh indeed, it is not used anymore I'll remove it.